modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/CursorResetHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.streaming;
+
+import org.mule.runtime.api.streaming.Cursor;
+import org.mule.runtime.api.util.Pair;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Entity in charge of reseting the position of a group of {@link Cursor}s .
+ *
+ * @since 4.3.1 4.4.0
+ */
+public class CursorResetHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CursorResetHandler.class);
+
+  private final List<Pair<Cursor, Long>> cursorPositions;
+
+  public CursorResetHandler(List<Pair<Cursor, Long>> cursorPositions) {
```

Comment:
this class should just receive the cursor. Clients are not responsible for building the data structures that it needs under hood. The list of pairs should be built here

Reply:
Agree, I have just changed this.